### PR TITLE
fix(runner): proper typing 

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

I got bite by the `any` currently returned by Temporal. 
Fix the types across the codebase and handle the error appropriately.

I havent touched a lot the runner so let me know if Im forgetting something

## Issue ticket number and link

Fixes NAN-338

### CODE REVIEW

File: packages/runner/lib/server.ts [hunk 1]
```diff
@@ -29,6 +30,7 @@ const appRouter = router({
 });
 
 export type AppRouter = typeof appRouter;
+export type ProxyAppRouter = CreateTRPCProxyClient<AppRouter>;
```
Comment (TBonnin) [inline packages/runner/lib/server.ts]: I think this type should be in `runner/lib/client.ts`

**Review by TBonnin — APPROVED** [review]
made one comment. Have you tested this in staging?

File: packages/cli/lib/services/local-integration.service.ts [hunk 2]
```diff
@@ -81,7 +82,15 @@ class IntegrationService implements IntegrationServiceInterface {
             } catch (err: any) {
                 // TODO merge this back with the main integration service
                 if (err instanceof ActionError) {
-                    return { success: false, error: err, response: null };
+                    return {
+                        success: false,
+                        error: {
+                            type: err.type,
+                            payload: err.payload || {},
+                            status: 500
```
Comment (khaliqgant) [inline packages/cli/lib/services/local-integration.service.ts:90]: is 500 what we always return? Doesn't NangoError default to 500?

Comment (bodinsamuel) [reply packages/cli/lib/services/local-integration.service.ts:90]: Yes it's the default, unless the type is known. 
I just followed the pattern in Line 80, which is returning 500 by default. Is it okay?

**Comment by bodinsamuel** [comment]
> Have you tested this in staging?

I didn't, I need to deploy Runner and jobs right?

**Comment by TBonnin** [comment]
> > Have you tested this in staging?
> 
> I didn't, I need to deploy Runner and jobs right?

yep, runner and jobs

**Comment by bodinsamuel** [comment]
Both worked 👌🏻

File: packages/jobs/lib/runner/render.runner.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import type { Runner } from './runner.js';
 import { RunnerType } from './runner.js';
-import { getRunnerClient } from '@nangohq/nango-runner';
+import { ProxyAppRouter, getRunnerClient } from '@nangohq/nango-runner';
```
Comment (khaliqgant) [inline packages/jobs/lib/runner/render.runner.ts:3]: `@nangohq/nango-runner` This doesn't exist? and won't resolve.

File: packages/jobs/lib/runner/runner.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import type { KVStore } from '@nangohq/shared/lib/utils/kvstore/KVStore.js';
 import { LocalRunner } from './local.runner.js';
 import { RenderRunner } from './render.runner.js';
 import { getEnv, getRedisUrl, InMemoryKVStore, RedisKVStore } from '@nangohq/shared';
+import type { ProxyAppRouter } from '@nangohq/nango-runner';
```
Comment (khaliqgant) [inline packages/jobs/lib/runner/runner.ts:5]: Same issue here.

**Pull request merged**